src/dmd/cppmanglewin.d

```diff
@@ -663,49 +663,268 @@ private:
+            assert(td);
+            assert(ti.tiargs.dim >= 1);
+            TemplateParameter tp = (*td.parameters)[0];
+            TemplateValueParameter tv = tp.isTemplateValueParameter();
+            if (!tv || !tv.valType.isString())
+                return false; // expecting a string argument to operators!
+            Expression exp = (*ti.tiargs)[0].isExpression();
+            StringExp str = exp.toStringExp();
+            final switch (whichOp)
+            {
+            case CppOperator.Unary:
+                switch (str.peekSlice())
+                {
+                    case "*":   symName = "?D";     goto continue_template;
+                    case "++":  symName = "?E";     goto continue_template;
+                    case "--":  symName = "?F";     goto continue_template;
+                    case "-":   symName = "?G";     goto continue_template;
+                    case "+":   symName = "?H";     goto continue_template;
+                    case "~":   symName = "?S";     goto continue_template;
+                    default:    return false;
+                }
+            case CppOperator.Binary:
+                switch (str.peekSlice())
+                {
+                    case ">>":  symName = "?5";     goto continue_template;
+                    case "<<":  symName = "?6";     goto continue_template;
+                    case "*":   symName = "?D";     goto continue_template;
+                    case "-":   symName = "?G";     goto continue_template;
+                    case "+":   symName = "?H";     goto continue_template;
+                    case "&":   symName = "?I";     goto continue_template;
+                    case "/":   symName = "?K";     goto continue_template;
+                    case "%":   symName = "?L";     goto continue_template;
+                    case "^":   symName = "?T";     goto continue_template;
+                    case "|":   symName = "?U";     goto continue_template;
+                    default:    return false;
+                    }
+            case CppOperator.OpAssign:
+                switch (str.peekSlice())
+                {
+                    case "*":   symName = "?X";     goto continue_template;
+                    case "+":   symName = "?Y";     goto continue_template;
+                    case "-":   symName = "?Z";     goto continue_template;
+                    case "/":   symName = "?_0";    goto continue_template;
+                    case "%":   symName = "?_1";    goto continue_template;
+                    case ">>":  symName = "?_2";    goto continue_template;
+                    case "<<":  symName = "?_3";    goto continue_template;
+                    case "&":   symName = "?_4";    goto continue_template;
+                    case "|":   symName = "?_5";    goto continue_template;
+                    case "^":   symName = "?_6";    goto continue_template;
+                    default:    return false;
+                }
```

Comment:
Did you look at the POSIX mangling by any chance ? Just curious to know if we can refactor common code for operators in their own, shared function (In a later PR).

Reply:
No, I've no idea. Manu probably knows if anyone, he wrote that, I just got rid of the cascaded `break`s.

Reply:
I have now, heh. Manu's got basically the same code in the posix, replete with cascading `break`s.

Reply:
;) .. Yeah they didn't actually start the same... but they drifted towards each other.